### PR TITLE
add developer guide for winery

### PR DESCRIPTION
**Pull request by mkepp**
added graphics for developer guide

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for UI changes)

### CODE REVIEW

File: docs/DeveloperGuide_Winery.md [hunk 1]
```diff
@@ -0,0 +1,622 @@
+---
```
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: Why four `---`?

Follwing header should be enough:

```
---
---
```

File: docs/DeveloperGuide_Winery.md [hunk 2]
```diff
@@ -0,0 +1,622 @@
...
+{}
+---
+---
+---
+<center>
+
+![alt text](graphics/Figure00-Cloudcycle.png "Cloudcyle")
```
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: Do not use "alt text", do the real description. Move `"Cloudcycle"`  to "alt text" and write it "CloudCycle".

File: docs/DeveloperGuide_Winery.md [hunk 3]
```diff
@@ -0,0 +1,622 @@
...
+
+###Bereitstellung, Verwaltung und vermarktung von portablen Cloud-Diensten mit  
+###garantierter Sicherheit und Compliance während des gesamten lebenszyklus
+
+</center>
+<center>
+
+#Winery - Developer Guide
+##Version 1.0
```
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: Do not put versions in markdown files

File: docs/DeveloperGuide_Winery.md [hunk 4]
```diff
@@ -0,0 +1,622 @@
+---
+{}
+---
+---
+---
+<center>
+
+![alt text](graphics/Figure00-Cloudcycle.png "Cloudcyle")
+
+###Bereitstellung, Verwaltung und vermarktung von portablen Cloud-Diensten mit
```
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: Use `<bold>`, not markdown headings. Markdown headings have different semantics.

File: docs/DeveloperGuide_Winery.md (hunk 2 above)
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: (in all other cases)

Here, remove the CloudCycle logo completely.

File: docs/DeveloperGuide_Winery.md (hunk 4 above)
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: (in all other cases)

here, remove the link to cloudcycle completely.

File: docs/DeveloperGuide_Winery.md [hunk 5]
```diff
@@ -0,0 +1,622 @@
...
+
+#Winery - Developer Guide
+##Version 1.0
+
+</center>
+
+---
+
+|                     |                                        | 
+|---------------------|----------------------------------------|
```
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: Remove the whole table.

In Git, version control works different from Word version control.

File: docs/DeveloperGuide_Winery.md [hunk 6]
```diff
@@ -0,0 +1,622 @@
...
+---
+
+#### Overview
+
+This document provides the design ideas of Winery.
+
+Winery is a Web based environment ro graphically model TOSCA topologies and plans managing these
+topologies. It is an Eclipse project and thus support is available through its project page
+[https://projects.eclipse.org/projects/soa.winery] (https://projects.eclipse.org/projects/soa.winery). Winery is also part of the OpenTOSCA ecosystem where more
+information is available at [http://www.opentosca.org](http://www.opentosca.org).
```
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: Hyperlinks are written <http://example.com/like-this> in markdown. In GitHub, you can use the plain link. No need to bouble encode.

File: docs/DeveloperGuide_Winery.md [hunk 7]
```diff
@@ -0,0 +1,622 @@
...
+The repository stores TOSCA models and allows managing their content. For instance, node types, policy types,
+and artifact templates are managed by the repository. The repository is also responsible for importing and 
+exporting CSARs, the exchange format of TOSCA files and related artifacts.
+
+---
+
+#### Code Location
+
+Winery is an Eclipse hosted project. Therefore, the code is versioned using git. One can browse the repository
+at [http://git.eclipse.org/c/winery/org.eclipse.winery.git/](http://git.eclipse.org/c/winery/org.eclipse.winery.git/). In the root of the repository, a README.md
```
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: Please fix the URL.

File: docs/DeveloperGuide_Winery.md [hunk 8]
```diff
@@ -0,0 +1,622 @@
...
+[http://git.eclipse.org/c/winery/org.eclipse.winery.git/plain/README.md](http://git.eclipse.org/c/winery/org.eclipse.winery.git/plain/README.md) is describing the inital setup in
```
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: Please fix the URL.

File: docs/DeveloperGuide_Winery.md [hunk 9]
```diff
@@ -0,0 +1,622 @@
...
+This document provides the design ideas of Winery.
+
+Winery is a Web based environment ro graphically model TOSCA topologies and plans managing these
+topologies. It is an Eclipse project and thus support is available through its project page
+[https://projects.eclipse.org/projects/soa.winery] (https://projects.eclipse.org/projects/soa.winery). Winery is also part of the OpenTOSCA ecosystem where more
+information is available at [http://www.opentosca.org](http://www.opentosca.org).
+
+<center>
+
+![alt text](graphics/Figure01-ComponentsOfWinery.png )
```
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: One space too much at the end?

File: docs/DeveloperGuide_Winery.md [hunk 10]
```diff
@@ -0,0 +1,622 @@
...
+This project contains the generator which generates a NodeType implementation based on a NodeType
+interface.
+
+##### Project org.eclipse.winery.model.csar.toscametafile
+
+This project contains the model for the file "Tosca.meta".
+
+##### Project org.eclipse.winery.model.selfservice
+
+This project contains the model for the self service portal. It is used by the Vinothek [BBKL14] to display user
```
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: Can you reformat the text to have one sentence per line?

Sentence 1 with words.
Sentence 2 with words.
Sentence 3 with words.
etc

File: docs/DeveloperGuide_Winery.md [hunk 11]
```diff
@@ -0,0 +1,622 @@
...
+Figure 3 shows the inheritance hierarchy of AdminId. "NamespacesId" is used as container for a mapping file
+from namespace prefixes to namespaces. "TypesId" is the parent element of all types user can set. This are 
+not node types etc., but ConstraintTypes (for Constraints), PlanLanguages (for plans), and PlanTypes (for plans).
+The inclusion of "PlanLanguages" is due to reuse of the class AbstractTypesManager for plan languages. TOSCA
+does not restrict these enumerations. Therefore, Winery manages all known types for itself.
+
+<center>
+
+![alt text](graphics/Figure03-IneritanceHierarchyofAdminId.png "Figure 3: Inheritance hierarchy of AdminId")  
+Figure 3: Inheritance hierarchy of AdminId
```
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: Please use the same format for captions throughout the document. Above, it was bold, how it is plain text.

File: docs/DeveloperGuide_Winery.md [hunk 12]
```diff
@@ -0,0 +1,622 @@
...
+
+#### References
+
+[BBKL14] Breitenbücher, Uwe; Binz, Tobias; Kopp, Oliver; Leymann, Frank: Vinothek - A Self-Service Portal for TOSCA. In: Herzberg, Nico (Hrsg); Kunze, Matthias (Hrsg): Proceedings of the 6th Central-European Workshop on Services and their Composition (ZEUS 2014).
+
+[KBBL12] Kopp, Oliver; Binz, Tobias; Breitenbücher, Uwe; Leymann, Frank: BPMN4TOSCA: A Domain-Specific Language to Model Management Plans for Composite Applications. In: Mendling, Jan (Hrsg); Weidlich, Matthias (Hrsg): 4th International Workshop on the Business Process Model and Notation, 2012
+
+---
+
+#### List of Figures
```
Comment (koppor) [inline docs/DeveloperGuide_Winery.md]: Remove that. No need for a list of figures here.

File: docs/DeveloperGuide_Winery.md (hunk 1 above)
Comment (mkepp) [inline docs/DeveloperGuide_Winery.md]: done

File: docs/DeveloperGuide_Winery.md (hunk 2 above)
Comment (mkepp) [inline docs/DeveloperGuide_Winery.md]: removed CloudCycle reference completely

File: docs/DeveloperGuide_Winery.md (hunk 4 above)
Comment (mkepp) [inline docs/DeveloperGuide_Winery.md]: removed CloudCycle reference completely

File: docs/DeveloperGuide_Winery.md (hunk 3 above)
Comment (mkepp) [inline docs/DeveloperGuide_Winery.md]: removed Version

File: docs/DeveloperGuide_Winery.md (hunk 5 above)
Comment (mkepp) [inline docs/DeveloperGuide_Winery.md]: done

File: docs/DeveloperGuide_Winery.md (hunk 6 above)
Comment (mkepp) [inline docs/DeveloperGuide_Winery.md]: fixed

File: docs/DeveloperGuide_Winery.md (hunk 9 above)
Comment (mkepp) [inline docs/DeveloperGuide_Winery.md]: deleted typo

File: docs/DeveloperGuide_Winery.md (hunk 7 above)
Comment (mkepp) [inline docs/DeveloperGuide_Winery.md]: fixed

File: docs/DeveloperGuide_Winery.md (hunk 8 above)
Comment (mkepp) [inline docs/DeveloperGuide_Winery.md]: fixed

File: docs/DeveloperGuide_Winery.md (hunk 10 above)
Comment (mkepp) [inline docs/DeveloperGuide_Winery.md]: done

File: docs/DeveloperGuide_Winery.md (hunk 11 above)
Comment (mkepp) [inline docs/DeveloperGuide_Winery.md]: all captions bold now

File: docs/DeveloperGuide_Winery.md (hunk 12 above)
Comment (mkepp) [inline docs/DeveloperGuide_Winery.md]: removed

**Comment by koppor** [comment]
These are more than 250 lines of code (considering markdown as code), thus the normal IP process has to be done. The CQ is https://dev.eclipse.org/ipzilla/show_bug.cgi?id=13783.

**Comment by koppor** [comment]
- [ ] remove CloudCycle logo from the commit

**Comment by koppor** [comment]
After re-iterating on https://dev.eclipse.org/mhonarc/lists/winery-dev/msg00011.html, I decide NOT to include this in the source code of Eclipse Winery.

@mkepp Please check whether you have write access at https://github.com/eclipse/winery/wiki.

**Comment by koppor** [comment]
I missed a `git push -f` to mkepp. The real commit matching the files of CQ 13783 is 8cd41317bd7781fdcc11797fed504e3cdcf03bd8

**Pull request merged**